### DISCUSSION TITLE
Stef's Magritte Tutorial clarity reqd

### DISCUSSION THREAD

**Discussion by sanjayminni** [magritte-metamodel/magritte#188] [General]
From pg 16 of "meta data at work with Magritte - Tutorial", Stef 

how are these to be entered since both method codes are in the same object and methodName

MAAddress >> descriptionEndDate
<magritteDescription>
^ MADateDescription new
accessor: #endDate;
label: 'End Date';
addCondition: [ :value | value > Date today ];
beRequired;
yourself

MAAddress >> descriptionEndDate
<magritteDescription>
^ super descriptionContainer
addCondition: [ :object |
(object readUsing: self descriptionEndDate) >
(object readUsing: self descriptionStartDate)]
labelled: 'End date must be after start date';
yourself

**Comment by sanjayminni** [comment]
got this to work as follows as of now
Changed the second part of the code - i.e. to push the (validation) rules to
the container as follows

MAAddress >> descriptionContainer       "// (i.e. not descriptionEndDate) as in the document"
    <magritteContainer>                        "// (i.e. not <magritteDescription> as in the document"
    ^ super descriptionContainer
        addCondition: [ :object |
            (object readUsing: self descriptionEndDate) >
            (object readUsing: self descriptionStartDate)]
        labelled: 'End date must be after start date';
       yourself

@Ducasse: If you want I can send a PR but I really haven't done it before and I
dont have a GH client on my machine if thats reqd - i only interact thru
iceberg - if thats relevant

**Reply by seandenigris** [reply]
Ah, yes. Looks like a typo. Good catch!
